package/MDAnalysis/coordinates/chemfiles.py

```diff
@@ -381,3 +382,63 @@ def _topology_to_chemfiles(self, obj, n_atoms):
             topology.add_bond(bond.atoms[0].ix, bond.atoms[1].ix)
 
         return topology
+
+
+if HAS_CHEMFILES:
+    class ChemfilesPicklable(chemfiles.Trajectory):
+        """Chemfiles file object (read-only) that can be pickled.
+
+        This class provides a file-like object (as returned by
+        :class:`chemfiles.Trajectory`) that,
+        unlike standard Python file objects,
+        can be pickled. Only read mode is supported.
+
+        When the file is pickled, path, mode, and format of the file handle
+        are saved. On unpickling, the file is opened by path with mode,
+        and saved format.
+        This means that for a successful unpickle, the original file still has
+        to be accessible with its filename.
+
+        Note
+        ----
+        Upon pickling, the current frame is reset. `universe.trajectory[i]` has
+        to be used to return to its original frame.
+
+        Parameters
+        ----------
+        filename : str
+            a filename given a text or byte string.
+        mode : 'r' , optional
+            can only be 'r' for pickling.
```

Comment:
it do not look like you are checking this in `__getstate__`, so users could try `ChemfilesPicklable(file, 'w')` without reading the documentation and have it fail silently or strangely